src/reducers/entities/threads.ts

```diff
@@ -87,6 +87,53 @@ export const threadsInTeamReducer = (state: ThreadsState['threadsInTeam'] = {},
 };
 export const countsReducer = (state: ThreadsState['counts'] = {}, action: GenericAction) => {
     switch (action.type) {
+    case ThreadTypes.ALL_TEAM_THREADS_READ: {
+        const counts = state[action.data.team_id] ?? {unread_mentions_per_channel: {}};
+        return {
+            ...state,
+            [action.data.team_id]: {
+                ...counts,
+                unread_mentions_per_channel: {},
+                total_unread_mentions: 0,
+            },
+        };
+    }
+    case ThreadTypes.READ_CHANGED_THREAD: {
+        const {channelId, newUnreadMentions, prevUnreadMentions, teamId} = action.data;
+        const counts = state[teamId] ? {
+            ...state[teamId],
+        } : {
+            unread_mentions_per_channel: {
+                [channelId]: 0,
+            },
+            total_unread_threads: 0,
+            total: 0,
+            total_unread_mentions: 0,
+        };
+        const unreadMentionDiff = newUnreadMentions - prevUnreadMentions;
+        if (counts.unread_mentions_per_channel[channelId]) {
+            const nc = {...counts.unread_mentions_per_channel};
+            nc[channelId] += unreadMentionDiff;
+            counts.unread_mentions_per_channel = nc;
+        } else {
+            counts.unread_mentions_per_channel = {[channelId]: newUnreadMentions};
+        }
+        counts.total_unread_mentions = (counts.total_unread_mentions ?? 0) + unreadMentionDiff;
+
+        if (newUnreadMentions === 0 && prevUnreadMentions > 0) { // thread was marked read, reduce global count
+            counts.total_unread_threads = (counts.total_unread_threads || 1) - 1;
+        }
+        if (newUnreadMentions > 0 && prevUnreadMentions === 0) { // new mentions appear, incread global count if needed
+            counts.total_unread_threads = (counts.total_unread_threads || 0) + 1;
+        }
```

Comment:
I think this logic will only apply when there are mentions, excluding changes to a thread's `unread_replies`.